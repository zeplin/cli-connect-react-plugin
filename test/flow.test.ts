import Plugin from "../src";

describe("Connected Components React Plugin - Flow", () => {
    test("FlowComponent.jsx snippet creation", async () => {
        const processor = new Plugin();

        const componentCode = await processor.process(
            {
                path: "test/samples/flow/FlowComponent.jsx",
                zeplinNames: []
            }
        );

        expect(componentCode).toMatchSnapshot();
    });

    test("FlowComponentWithProps.jsx snippet creation", async () => {
        const processor = new Plugin();

        const componentCode = await processor.process(
            {
                path: "test/samples/flow/FlowComponentWithProps.jsx",
                zeplinNames: []
            }
        );

        expect(componentCode).toMatchSnapshot();
    });

    test("FlowComponentWithChildren.jsx snippet creation", async () => {
        const processor = new Plugin();

        const componentCode = await processor.process(
            {
                path: "test/samples/flow/FlowComponentWithChildren.jsx",
                zeplinNames: []
            }
        );

        expect(componentCode).toMatchSnapshot();
    });

    test("FlowComponentWithChildrenAndProps.jsx snippet creation", async () => {
        const processor = new Plugin();

        const componentCode = await processor.process(
            {
                path: "test/samples/flow/FlowComponentWithChildrenAndProps.jsx",
                zeplinNames: []
            }
        );

        expect(componentCode).toMatchSnapshot();
    });
});
