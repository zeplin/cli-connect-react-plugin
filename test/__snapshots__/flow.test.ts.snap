// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connected Components React Plugin - Flow FlowComponent.jsx snippet creation 1`] = `
Object {
  "description": "General component description.",
  "lang": "jsx",
  "snippet": "<MyComponent />",
}
`;

exports[`Connected Components React Plugin - Flow FlowComponentWithChildren.jsx snippet creation 1`] = `
Object {
  "description": "General component description.",
  "lang": "jsx",
  "snippet": "<MyComponent>
  {children}
</MyComponent>",
}
`;

exports[`Connected Components React Plugin - Flow FlowComponentWithChildrenAndProps.jsx snippet creation 1`] = `
Object {
  "description": "General component description.",
  "lang": "jsx",
  "snippet": "<MyComponent
  primitive={number}
  literalsAndUnion={'string' | 'otherstring' | number}
  arr={Array<any>}
  func={(value: string) => void}
  noParameterName={string => void}
  obj={{ subvalue: boolean }}>
  {children}
</MyComponent>",
}
`;

exports[`Connected Components React Plugin - Flow FlowComponentWithProps.jsx snippet creation 1`] = `
Object {
  "description": "General component description.",
  "lang": "jsx",
  "snippet": "<MyComponent
  primitive={number}
  literalsAndUnion={'string' | 'otherstring' | number}
  arr={Array<any>}
  func={(value: string) => void}
  noParameterName={string => void}
  obj={{ subvalue: boolean }} />",
}
`;

exports[`Connected Components React Plugin - Flow MultiExportFlowComponentWithProps.jsx snippet creation 1`] = `
Object {
  "description": "Component 1 description. Only this one should be shown",
  "lang": "jsx",
  "snippet": "<MyComponent
  primitive={number}
  literalsAndUnion={'string' | 'otherstring' | number}
  arr={Array<any>}
  func={(value: string) => void}
  noParameterName={string => void}
  obj={{ subvalue: boolean }} />

<MyOtherComponent
  primitive={number}
  literalsAndUnion={'string' | 'otherstring' | number}
  arr={Array<any>} />",
}
`;
